.jenkins/cscs/env-common.sh

```diff
@@ -8,7 +8,7 @@ module load daint-gpu spack-config
 
 export SPACK_ROOT="/apps/daint/SSL/pika/spack"
 set +u
-test -z "$SPACK_USER_CONFIG_PATH" && export SPACK_USER_CONFIG_PATH="${SPACK_ROOT}/../spack-user-config"
+export SPACK_USER_CONFIG_PATH="${SPACK_ROOT}/../spack-user-config"
```

Comment:
Move this outside the `set +u` (just for style, it doesn't have a practical difference since `SPACK_ROOT` is definitely set). Feel free to merge after that.